### PR TITLE
Add table aws_redshift_subnet_group closes #198

### LINKED ISSUE #198: Add table aws_redshift_subnet_group
**References**
https://pkg.go.dev/github.com/aws/aws-sdk-go@v1.37.24/service/redshift/redshiftiface

### PR DESCRIPTION
**Pull request by Paulami30**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
No env file present for the current environment:  staging 
 Falling back to .env config
No env file present for the current environment:  staging
customEnv TURBOT_TEST_EXPECTED_TIMEOUT undefined

SETUP: tests/aws_redshift_subnet_group []

PRETEST: tests/aws_redshift_subnet_group

TEST: tests/aws_redshift_subnet_group
Running terraform
data.aws_region.alternate: Refreshing state...
data.aws_region.primary: Refreshing state...
data.aws_partition.current: Refreshing state...
data.aws_caller_identity.current: Refreshing state...
data.null_data_source.resource: Refreshing state...
aws_vpc.my_vpc: Creating...
aws_vpc.my_vpc: Still creating... [10s elapsed]
aws_vpc.my_vpc: Still creating... [20s elapsed]
aws_vpc.my_vpc: Creation complete after 29s [id=vpc-0ccd35fe4efc95678]
aws_subnet.my_subnet: Creating...
aws_subnet.my_subnet: Still creating... [10s elapsed]
aws_subnet.my_subnet: Creation complete after 15s [id=subnet-087088805169b03a0]
aws_redshift_subnet_group.named_test_resource: Creating...
aws_redshift_subnet_group.named_test_resource: Still creating... [10s elapsed]
aws_redshift_subnet_group.named_test_resource: Creation complete after 15s [id=turbottest57717]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Apply complete! Resources: 3 added, 0 changed, 0 destroyed.

Outputs:

resource_aka = arn:aws:redshift:us-east-1:352685002396:subnetgroup:turbottest57717
resource_id = turbottest57717
resource_name = turbottest57717
vpc_id = vpc-0ccd35fe4efc95678

Running SQL query: test-get-query.sql
[
  {
    "cluster_subnet_group_name": "turbottest57717",
    "description": "A test subnet group",
    "subnet_group_status": "Complete",
    "tags_src": [
      {
        "Key": "name",
        "Value": "turbottest57717"
      }
    ],
    "vpc_id": "vpc-0ccd35fe4efc95678"
  }
]
✔ PASSED

Running SQL query: test-hydrate-query.sql
[
  {
    "akas": [
      "arn:aws:redshift:us-east-1:352685002396:subnetgroup:turbottest57717"
    ],
    "cluster_subnet_group_name": "turbottest57717",
    "description": "A test subnet group",
    "tags": {
      "name": "turbottest57717"
    },
    "title": "turbottest57717"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "akas": [
      "arn:aws:redshift:us-east-1:352685002396:subnetgroup:turbottest57717"
    ],
    "cluster_subnet_group_name": "turbottest57717",
    "description": "A test subnet group",
    "subnet_group_status": "Complete",
    "tags": {
      "name": "turbottest57717"
    },
    "title": "turbottest57717"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql
null
✔ PASSED

POSTTEST: tests/aws_redshift_subnet_group

TEARDOWN: tests/aws_redshift_subnet_group

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
> select
  cluster_subnet_group_name,
  description,
  subnet_group_status,
  vpc_id
from
  aws_redshift_subnet_group;
+---------------------------+-------------+---------------------+-----------------------+
| cluster_subnet_group_name | description | subnet_group_status | vpc_id                |
+---------------------------+-------------+---------------------+-----------------------+
| cluster-subnet-group-3    | test        | Complete            | vpc-072b70de006ffe88b |
+---------------------------+-------------+---------------------+-----------------------+

> select
  cluster_subnet_group_name,
  subnet -> 'SubnetAvailabilityZone' ->> 'Name' as subnet_availability_zone,
  subnet -> 'SubnetAvailabilityZone' ->> 'SupportedPlatforms' as supported_platforms,
  subnet ->> 'SubnetIdentifier' as subnet_identifier,
  subnet ->> 'SubnetStatus' as subnet_status
from
  aws_redshift_subnet_group,
  jsonb_array_elements(subnets) as subnet;
+---------------------------+--------------------------+---------------------+--------------------------+---------------+
| cluster_subnet_group_name | subnet_availability_zone | supported_platforms | subnet_identifier        | subnet_status |
+---------------------------+--------------------------+---------------------+--------------------------+---------------+
| cluster-subnet-group-3    | us-east-1e               | <null>              | subnet-0ff7d7ff93f78f4ed | Active        |
+---------------------------+--------------------------+---------------------+--------------------------+---------------+

> select
  cluster_subnet_group_name,
  tags
from
  aws_redshift_subnet_group
where
  not tags :: JSONB ? 'application';
+---------------------------+----------------+
| cluster_subnet_group_name | tags           |
+---------------------------+----------------+
| cluster-subnet-group-3    | {"test":"dev"} |
+---------------------------+----------------+
```
</details>

### CODE REVIEW

File: aws/table_aws_redshift_subnet_group.go [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+			{
+				Name:        "description",
+				Description: "The description of the cluster subnet group.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "subnet_group_status",
+				Description: "The status of the cluster subnet group.",
+				Type:        proto.ColumnType_STRING,
+			},
```
Comment (Subhajit97) [inline aws/table_aws_redshift_subnet_group.go:44]: Add `subnet_group_status` column before `description`.

File: aws/table_aws_redshift_subnet_group.go [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+				Description: resourceInterfaceDescription("tags"),
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.From(getRedshiftSubnetGroupTurbotTags),
+			},
+			{
+				Name:        "title",
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("ClusterSubnetGroupName"),
+			},
```
Comment (Subhajit97) [inline aws/table_aws_redshift_subnet_group.go:75]: Move `tags` column after `title`.
Generally we add the standard column in this order(i.e. `title`, `tags` and `akas`)

File: aws/table_aws_redshift_subnet_group.go [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+	// Get the resource tags
+	var turbotTagsMap map[string]string
+	if data.Tags != nil {
+		turbotTagsMap = map[string]string{}
+		for _, i := range data.Tags {
+			turbotTagsMap[*i.Key] = *i.Value
+		}
+	}
+	return turbotTagsMap, nil
+}
```
Comment (Subhajit97) [inline aws/table_aws_redshift_subnet_group.go:165]: Move this code block after `getRedshiftSubnetGroupAkas`, as we add the tarnsform functions at the end

File: aws-test/tests/aws_redshift_subnet_group/variables.tf [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  cidr_block        = "10.0.0.0/24"
+  availability_zone = "${var.aws_region}a"
+}
+
+resource "aws_redshift_subnet_group" "named_test_resource" {
+  name        = var.resource_name
+  description = "A test subnet group"
+  subnet_ids = [
+    "${aws_subnet.my_subnet.id}",
+  ]
```
Comment (Subhajit97) [inline aws-test/tests/aws_redshift_subnet_group/variables.tf:64]:
```suggestion
  subnet_ids = [aws_subnet.my_subnet.id]
```

**Review by Subhajit97 — CHANGES_REQUESTED** [review]
Make suggested changes

File: aws/table_aws_redshift_subnet_group.go [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+
+	// Get data for turbot defined properties
+	akas := []string{arn}
+
+	return akas, nil
+}
+
+//// TRANSFORM FUNCTIONS
+
+func getRedshiftSubnetGroupTurbotTags(_ context.Context, d *transform.TransformData) (interface{}, error) {
```
Comment (Subhajit97) [inline aws/table_aws_redshift_subnet_group.go:168]:
```suggestion
func redshiftSubnetGroupTurbotTags(_ context.Context, d *transform.TransformData) (interface{}, error) {
```

File: aws/table_aws_redshift_subnet_group.go [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+	data := d.HydrateItem.(*redshift.ClusterSubnetGroup)
+	if data.Tags == nil {
+		return nil, nil
+	}
+	// Get the resource tags
```
Comment (Subhajit97) [inline aws/table_aws_redshift_subnet_group.go:173]:
```suggestion

	// Get the resource tags
```

File: docs/tables/aws_redshift_subnet_group.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+```sql
+select
+  cluster_subnet_group_name,
+  subnet -> 'SubnetAvailabilityZone' ->> 'Name' as subnet_availability_zone,
+  subnet -> 'SubnetAvailabilityZone' ->> 'SupportedPlatforms' as supported_platforms,
+  subnet ->> 'SubnetIdentifier' as subnet_identifier,
+  subnet ->> 'SubnetStatus' as subnet_status
+from
+  aws_redshift_subnet_group
+  cross join jsonb_array_elements(subnets) as subnet;
```
Comment (Subhajit97) [inline docs/tables/aws_redshift_subnet_group.md:31]: avoid using `cross join` when you don't need it in the examples
```suggestion
  aws_redshift_subnet_group,
  jsonb_array_elements(subnets) as subnet;
```

**Review by Subhajit97 — CHANGES_REQUESTED** [review]
@Paulami30 There are few more changes required, otherwise it looks good.

**Review by Subhajit97 — APPROVED** [review]
LGTM

**Review by rajmohanty17 — APPROVED** [review]
LGTM

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**